File: week1ButItsJava/src/TimerAgent.java
import jade.core.Agent;
import jade.core.behaviours.TickerBehaviour;

public class TimerAgent extends Agent{
    int w = 15;
    public void setup(){
        addBehaviour(new TickerBehaviour(this, 1000) {
            @Override
            protected void onTick() {
                if(w>0){
                    System.out.println(w+ " seconds left.");
                    w--;
                }else {
                    System.out.println("byeee");
                    myAgent.doDelete();
                }
            }
        });
    }
}

File: week1/src/TimerAgent.kt
import jade.core.Agent
import jade.core.behaviours.TickerBehaviour

class TimerAgent : Agent(){
    var w = 15
    override fun setup() {
        addBehaviour(object: TickerBehaviour(this,1000){
            override fun onTick(){
                if(w >0){
                    println("$w seconds left.")
                    w-=1
                }else{
                    println("byee")
                    myAgent.doDelete()
                }
            }
        })
    }
}